### ISSUE TITLE
patch failure message

### ISSUE THREAD

**Issue by aawce** [ikwzm/zynqmp-gpu-kmod-dpkg#1]
Hi,
I have downloaded and followed the instructions for this zynqmp mali driver, but I am seeing an error message in applying the patch (see below) which is different than that reported in readme.md.

Perhaps the xilinx patches have changed?  Should I dowload a specific xilinx tag?  FYI I am targeting 2019.2 xlnx kernel using Ubuntu 18.04.4 OS.

Please advise.
thks,
Chris

patching file Makefile
patching file platform/arm/arm.c
patching file linux/mali_linux_trace.h
patching file platform/arm/arm.c
patching file linux/mali_kernel_linux.c
patching file platform/arm/arm.c
patching file common/mali_pm.c
patching file linux/mali_kernel_linux.c
patching file linux/mali_memory_os_alloc.c
Hunk #1 FAILED at 202.
1 out of 1 hunk FAILED -- saving rejects to file linux/mali_memory_os_alloc.c.rej
patching file linux/mali_memory_secure.c
patching file common/mali_control_timer.c
patching file common/mali_group.c
patching file common/mali_osk.h
patching file linux/mali_osk_timers.c
patching file linux/mali_memory_os_alloc.c
Hunk #1 succeeded at 235 (offset -4 lines).
patching file linux/mali_ukk_mem.c
patching file linux/mali_memory_block_alloc.c
patching file linux/mali_memory_cow.c
patching file linux/mali_memory_os_alloc.c
Hunk #1 succeeded at 372 (offset -6 lines).
Hunk #2 succeeded at 415 (offset -6 lines).
Hunk #3 succeeded at 442 (offset -6 lines).
patching file linux/mali_memory_secure.c
patching file linux/mali_memory.c
patching file linux/mali_osk_time.c

**Comment by ikwzm** [comment]
Thank you for the issue.

As you pointed out, meta-xilinx was updated.
I used 391c7054e88ae77abf18fe8a705ac7ff34c7dc79.

```
shell$ git clone https://github.com/Xilinx/meta-xilinx.git
Cloning into 'meta-xilinx'...
remote: Enumerating objects: 840, done.        
remote: Counting objects: 100% (840/840), done.        
remote: Compressing objects: 100% (395/395), done.        
remote: Total 11356 (delta 474), reused 732 (delta 404), pack-reused 10516        
Receiving objects: 100% (11356/11356), 8.76 MiB | 6.08 MiB/s, done.
Resolving deltas: 100% (6244/6244), done.

shell$ git checkout 391c7054e88ae77abf18fe8a705ac7ff34c7dc79
Note: checking out '391c7054e88ae77abf18fe8a705ac7ff34c7dc79'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 391c705 meta-xilinx-contrib: Fix drm patch for v4.19 kernel
```

**Comment by aawce** [comment]
Thanks!  FYI since I'm targeting 2019.2, I did the following successfully with the different recommended versions:

> git clone https://github.com/Xilinx/meta-xilinx.git
> cd meta-xilinx/;  git checkout remotes/origin/rel-v2019.2;    cd ../
> wget https://developer.arm.com/-/media/Files/downloads/mali-drivers/kernel/mali-utgard-gpu/DX910-SW-99002-r9p0-01rel0.tgz
> tar xzf DX910-SW-99002-r9p0-01rel0.tgz
> source patch.sh
patching file Makefile
patching file platform/arm/arm.c
patching file linux/mali_linux_trace.h
patching file platform/arm/arm.c
patching file linux/mali_kernel_linux.c
patching file platform/arm/arm.c
patching file common/mali_pm.c
patching file linux/mali_kernel_linux.c
patching file linux/mali_memory_os_alloc.c
patching file linux/mali_memory_secure.c
patching file common/mali_control_timer.c
patching file common/mali_group.c
patching file common/mali_osk.h
patching file linux/mali_osk_timers.c
patching file linux/mali_memory_os_alloc.c
Hunk #1 succeeded at 241 (offset 2 lines).

Then edit debian/rules to point to new subdir DX910-SW-99002-r9p0-01rel0, then sudo debian/rules binary
